modules/xmpp/JingleSessionPC.js

```diff
@@ -1995,6 +1852,14 @@ export default class JingleSessionPC extends JingleSession {
                 logger.debug(`Existing SSRC re-mapped ${ssrc}: new owner=${owner}, source-name=${source}`);
 
                 this._signalingLayer.setSSRCOwner(ssrc, owner, source);
+                const oldSourceName = track.getSourceName();
+                const sourceInfo = this.peerconnection.getRemoteSourceInfoBySourceName(oldSourceName);
+
+                // Update the SSRC map on the peerconnection.
+                if (sourceInfo) {
+                    this.peerconnection._remoteSsrcMap.delete(oldSourceName);
```

Comment:
Having to dig in the private methods here looks nasty shall this have some API?

Reply:
Oops, I added TPC.updateRemoteSources for this exact purpose later but forgot to update it here. Thanks for pointing it out.